projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -23,7 +35,14 @@ export class CartesianLegend {
     private readonly injector: Injector,
     private readonly intervalData?: CartesianIntervalData,
     private readonly summaries: Summary[] = []
-  ) {}
+  ) {
+    this.activeSeries = [...this.series];
```

Comment:
and here, we're duping the series because we mutate it above, which would otherwise effect here :) lots of side effects that different places need to be in sync with - better to not touch `this.series` as described above.